File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/DccProposalMapper.cs
using DccProposalSdk = OnlinePayments.Sdk.Domain.DccProposal;
using DccProposalDto = Business.Domain.Payments.PaymentDetails.DccProposal;

namespace Infrastructure.Mappers.PaymentDetails;

public static class DccProposalMapper
{
    public static DccProposalDto Map(DccProposalSdk? response)
    {
        return new()
        {
            Rate = RateDetailsMapper.Map(response?.Rate),
            BaseAmount = AmountOfMoneyMapper.Map(response?.BaseAmount),
            DisclaimerDisplay = response?.DisclaimerDisplay,
            DisclaimerReceipt = response?.DisclaimerReceipt,
            TargetAmount = AmountOfMoneyMapper.Map(response?.TargetAmount)
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/DccProposalMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\DccProposal as DccProposalDto;
use OnlinePayments\Sdk\Domain\DccProposal as DccProposalSdk;

final class DccProposalMapper
{
    public static function mapFromSdkResponse(?DccProposalSdk $response): ?DccProposalDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new DccProposalDto();
        $dto->rate = RateDetailsMapper::mapFromSdkResponse($response?->rate ?? null);
        $dto->baseAmount = AmountOfMoneyMapper::mapFromSdkResponse($response?->baseAmount ?? null);
        $dto->disclaimerDisplay = $response?->disclaimerDisplay ?? null;
        $dto->disclaimerReceipt = $response?->disclaimerReceipt ?? null;
        $dto->targetAmount = AmountOfMoneyMapper::mapFromSdkResponse($response?->targetAmount ?? null);

        return $dto;
    }
}
